tests/rptest/tests/datalake/datalake_e2e_test.py

```diff
@@ -191,14 +191,36 @@ def test_remove_expired_snapshots(self, cloud_storage_type):
                 "set tblproperties ('history.expire.max-snapshot-age-ms'='1000')"
             )
 
-            def has_one_snapshot():
+            def num_snapshots() -> int:
                 snapshots_out = spark.run_query_fetch_all(
                     f"select * from {table_name}.snapshots")
-                return len(snapshots_out) == 1
+                return len(snapshots_out)
 
-            wait_until(has_one_snapshot, timeout_sec=30, backoff_sec=1)
+            wait_until(lambda: num_snapshots() == 1,
+                       timeout_sec=30,
+                       backoff_sec=1)
             dl.wait_for_translation(self.topic_name, count)
 
+            # Externally create another snapshot. Redpanda should retain this
+            # snapshot _and_ the most recent one generated by Redpanda.
+            spark.make_client().cursor().execute(
+                f"insert into {table_name} (select * from {table_name} limit 1)"
+            )
+            num_snaps = num_snapshots()
+            assert num_snaps == 2, f"Expected 2 snapshots after writing: {num_snaps}"
+            time.sleep(10)
+            num_snaps = num_snapshots()
+            assert num_snaps == 2, f"Expected to retain 2 snapshots: {num_snaps}"
```

Comment:
nitpick: would it be a bit cleaner to generate a third snapshot, `assert num == 3`, then `wait_until num ==2`?

Reply:
Hm I'm not sure, I guess I'd worry it'd be flaky given how frequently we expire snapshots (ie in between creating the snapshot and the assertion we could remove down to 2).

Is there specific coverage you're looking to add with another snapshot?

Reply:
> specific coverage?

No not really. knee-jerk reaction to `time.sleep`. I don't think I'd grokked that we expire snapshots that aggressively - so the idea here is that 10s is generous by like several orders of magnitude?

Reply:
It's a couple commit intervals. I guess we don't expire snapshots _that_ aggressively, but there's always room for a race since it's hard to control when expiry runs in Redpanda from this test.